python/pyfunc-server/pyfuncserver/publisher/kafka.py

```diff
@@ -23,3 +23,4 @@ def produce(self, data: PyFuncOutput):
         prediction_log = build_prediction_log(pyfunc_output=data, model_manifest=self.model_manifest)
         serialized_data = prediction_log.SerializeToString() 
         self.producer.produce(topic=self.topic, value=serialized_data)
+        self.producer.poll(0)
```

Comment:
Why is this not required in the [merlin logger](https://github.com/caraml-dev/merlin/blob/9f3b161527caea636e5a5e3fed77b4b1d68cab99/api/pkg/kafka/kafka.go#L84) 

Is this because of the `"acks": publisher_config.kafka.acks,` above?

Reply:
I don't think so. I think this is due to golang client will create poll go routine when [initialize producer](https://github.com/confluentinc/confluent-kafka-go/blob/master/kafka/producer.go#L594) hence no need to call poll and no `poll` function from golang producer